text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+
+Each view has a unique name, which may be automatically determined from the metric instrument and aggregation.
+Implementations should refuse to register two views with the same name.
+
+Note that a view does not describe the type (e.g. `int`, `float`) or unit of measurement (e.g. "bytes", "milliseconds") of the metric to be exported.
+The unit is determined by the metric instrument, and the aggregation and exporter may preserve or change the unit.
+For example, a hypothetical _MinMaxSumMeanCount_ aggregation of an int-valued millisecond latency measure may be exported as five separate metrics to the APM backend:
+
+- An int-valued _min_ metric with "ms" units
+- An int-valued _max_ metric with "ms" units
+- An int-valued _sum_ metric with "ms" units
+- A **float**-valued _mean_ metric with "ms" units
+- An **int** valued, unitless (i.e. unit "1") _count_ metric
+
+This OTEP doesn't propose a particular API for Aggregators, just that the API is sufficient for exporters to get all this information, including:
+
+- That the _min_, _max_, and _sum_ metrics preserve the type and unit of the underlying Measure
+- That the _mean_ metric is float-valued, but preserves the underlying measure's unit
+- That the _count_ metric is int-valued with unit "1", regardless of the underlying measure's unit
+
+### Aggregating over time
+
+An aggregation describes how multiple measurements captured via the same metric instrument in a single collection interval are combined.
+_Aggregators_ are SDK objects, and the default SDK includes an Aggregator interface (see [opentelemetry-specification#347](https://github.com/open-telemetry/opentelemetry-specification/pull/347) for details).
+
+Aggregations are assumed to be [mergeable](https://www.cs.utah.edu/~jeffp/papers/mergeSumm-MASSIVE11.pdf): aggregating a sequence of measurements should produce the same result as partitioning the sequence into subsequences, aggregating each subsequence, and combining the results.
+
+Said differently: given an aggregation function `agg` and sequence of measurements `S`, there should exist a function `merge` such that:
+
+```
+agg(S) = merge(agg(S_1), agg(S_2), ..., agg(S_N))
+```
+
+For every partition `{S_1, ..., S_N}` of `S`.
+
+For example, the _min_ aggregation is trivially mergeable:
+
+```
+min([1, 2, 3, 4, 5, 6]) = min([min([1]), min([2, 3]), min([4, 5, 6])])
+```
+
+but quantile aggregations, e.g. _p95_ and _p99_ are not.
+Applications that export quantile metrics should use a mergeable aggregations such as [DDSketch](https://arxiv.org/abs/1908.10693), which estimates quantile values with bounded errors, or export raw measurements without aggregation and compute exact quantiles on the backend.
+
+We require aggregations to be mergeable so that they produce the same results regardless of the collection interval, or the number of collection events per export interval.
```

Comment:
I guess again looking at quantiles as the edge case: what's the need for ensuring that the produced result are the same, regardless of collection interval? Is there some caveat where the collection interval cannot be guaranteed? I guess in current API design it's in the pushController side of things.